forge-gui-desktop/src/main/java/forge/itemmanager/CardManager.java

```diff
@@ -92,6 +93,17 @@ protected Iterable<Entry<PaperCard, Integer>> getUnique(Iterable<Entry<PaperCard
         return cardsMap.entrySet();
     }
 
+    @Override
+    public void updateView(final boolean forceFilter, final Iterable<PaperCard> itemsToSelect) {
+        prefetchImages();
+        super.updateView(forceFilter, itemsToSelect);
+    }
+
+    private void prefetchImages() {
+        if (isInfinite() || getPool().isEmpty()) return;
+        GuiBase.getInterface().prefetchAllImages(getPool());
```

Comment:
I'm a little concerned about how this is fetching. Thats potentially a lot of downloads immediately, and woud be going against Scryfalls 10 requests per minute limitations. https://scryfall.com/docs/api

Reply:
Yea i mostly worry with all of the new contributors that this would get changed without anyone noticing if there's not explicit protections around it. That's part of the reason we wholesale removed the bulk downloaders when switching over to Scryfall download.

Reply:
Now rate-limiting to 10 req/sec using a `ScheduledExecutorService`. https://github.com/Card-Forge/forge/pull/7838/commits/1cab968e9404724c528d1fc33e5915a781284e1c